sky/clouds/azure.py

```diff
@@ -137,6 +137,51 @@ def get_egress_cost(self, num_gigabytes: float):
     def is_same_cloud(self, other):
         return isinstance(other, Azure)
 
+    @classmethod
+    def get_image_size(cls, image_id: str, region: Optional[str]) -> float:
+        if region is None:
+            # The region used here is only for where to send the query,
+            # not the image location. Azure's image is globally available.
+            region = 'eastus'
+        is_skypilot_image_tag = False
+        if image_id.startswith('skypilot:'):
+            is_skypilot_image_tag = True
+            image_id = service_catalog.get_image_id_from_tag(image_id,
+                                                             clouds='azure')
+        image_id_splitted = image_id.split(':')
+        if len(image_id_splitted) != 4:
+            with ux_utils.print_exception_no_traceback():
+                raise ValueError(f'Invalid image id: {image_id}. Expected '
+                                 'format: <publisher>:<offer>:<sku>:<version>')
+        publisher, offer, sku, version = image_id_splitted
+        if is_skypilot_image_tag:
+            if offer == 'ubuntu-hpc':
+                return _DEFAULT_AZURE_UBUNTU_HPC_IMAGE_GB
+            else:
+                assert offer == 'ubuntu-2004', offer
```

Comment:
Let's avoid the assertion here, so that customized image (user changes images.csv manually) will not cause an error.

Reply:
Good point! Removed.